### PR TITLE
Switch this charm to be a subordinate deployed against openstack-integrator

### PR DESCRIPTION
**Pull request by addyess**
This charm hasn't ever been published to charmhub.  So upgrades weren't something to be addressed

This helps the `cinder-csi` follow in-line with other charms like `aws-k8s-storage`, and `gcp-k8s-storage` which deploy against their "integrator" charm as a subordinate, and use the kubernetes credentials provided via `kube-control` and the api certs provided via `tls-interface` to apply a specific mutated upstream version of cinder-csi into the cluster.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
-ops >= 1.2.0
 lightkube>=0.10.1,<1.0.0
-serialized-data-interface>=0.4.0,<1.0.0
+ops>=1.3.0,<2.0.0
+ops.manifest>=1.1.0,<2.0.0
+pydantic
+git+https://github.com/charmed-kubernetes/interface-kube-control.git@6dd289d1c795fdeda1bed17873b8d6562227c829#subdirectory=ops
+git+https://github.com/juju-solutions/interface-openstack-integration.git@0f81bb414ad8676ab889bbba24a34cffd12ce2ca#subdirectory=ops
```
Comment (addyess) [inline requirements.txt]: DO NOT MERGE with this requirement -- ensure it's from charmed-kubernetes main branch commit

File: metadata.yaml [hunk 2]
```diff
@@ -1,22 +1,25 @@
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
 
-name: cinder-csi-operator
+name: cinder-csi
 display-name: Cinder CSI Operator
-description: Deploys and manages the Cinder CSI plugin for K8s on OpenStack.
 summary: Deploys and manages the Cinder CSI plugin for K8s on OpenStack.
+description: Deploys and manages the Cinder CSI plugin for K8s on OpenStack.
+docs: https://discourse.charmhub.io/t/cinder-csi-docs-index/????
```
Comment (mateoflorido) [inline metadata.yaml]: Shouldn't we create at least a placeholder in discourse?

File: src/charm.py [hunk 3]
```diff
@@ -4,74 +4,228 @@
...
+    def _install_or_upgrade(self, event, config_hash=None):
+        if self.stored.config_hash == config_hash:
+            log.info("Skipping until the config is evaluated.")
+            return True
 
-    def _set_version(self, event=None):
-        if self.unit.is_leader():
-            self.unit.set_workload_version(self.version)
+        self.unit.status = MaintenanceStatus("Deploying Cinder Storage")
+        self.unit.set_workload_version("")
```
Comment (mateoflorido) [inline src/charm.py:205]:
```suggestion
        self.unit.set_workload_version(self.collector.short_version)
```
I think the version is missing. Am I overlooking somethig here?

**Review by mateoflorido — APPROVED** [review]
LGTM, just a few nits

File: metadata.yaml (hunk 2 above)
Comment (addyess) [reply metadata.yaml]: awesome catch!  I meant to do that before finishing this PR. 
THANK YOU!

File: src/charm.py (hunk 3 above)
Comment (addyess) [reply src/charm.py:205]: that's done on the `_update_status` hook if the charm is successfully deployed.  The intent here is to clear it during an upgrade/install and set it after a successful install in `update_status`

**Review by mateoflorido — APPROVED** [review]
LGTM! 💯

**Pull request merged**